bot/orchestration/src/main/kotlin/ai/tock/bot/orchestration/bot/primary/PrimaryBotConfiguration.kt

```diff
@@ -22,16 +22,27 @@ import ai.tock.bot.definition.StoryDefinition
 import ai.tock.bot.engine.BotBus
 import ai.tock.bot.orchestration.shared.OrchestrationData
 import ai.tock.bot.orchestration.shared.OrchestrationTargetedBot
+import java.time.Duration
+import java.time.Instant
 
 data class PrimaryBotConfiguration(
-    val startOrchestrationIntentList : List<Intent> = emptyList(),
-    val stopOrchestrationIntentList : List<Intent> = emptyList(),
-    val noOrchestrationIntentList : List<Intent> = emptyList(),
-    val comebackStory : StoryDefinition,
+    val startOrchestrationIntentList: List<Intent> = emptyList(),
+    val stopOrchestrationIntentList: List<Intent> = emptyList(),
+    val noOrchestrationIntentList: List<Intent> = emptyList(),
+    @Deprecated("use PrimaryBotOrchestrationEventHandler.onStopOrchestration") val comebackStory: StoryDefinition,
     private val eligibleTargetBotsByConnector: Map<ConnectorType, List<OrchestrationTargetedBot>> = emptyMap(),
-    private val dataProvider : OrchestrationDataProvider = DefaultOrchestrationDataProvider()
+    private val dataProvider: OrchestrationDataProvider = DefaultOrchestrationDataProvider(),
+    val takeBackOrchestration: ((BotBus) -> Boolean)? = null,
+    val primaryBotOrchestrationEventHandler: PrimaryBotOrchestrationEventHandler = DefaultPrimaryBotOrchestrationEventHandler(comebackStory)
 ) {
-    fun getEligibleTargetBots(connectorType: ConnectorType) : List<OrchestrationTargetedBot> = eligibleTargetBotsByConnector[connectorType] ?: emptyList()
+    fun getEligibleTargetBots(connectorType: ConnectorType): List<OrchestrationTargetedBot> =
+        eligibleTargetBotsByConnector[connectorType] ?: emptyList()
 
-    fun getOrchestrationData(bus : BotBus) : OrchestrationData? = dataProvider.provideOrchestrationData(bus)
+    fun getOrchestrationData(bus: BotBus): OrchestrationData? = dataProvider.provideOrchestrationData(bus)
+
+    companion object {
+        fun takeBackOrchestrationByTimeOut(inactivityDuration: Duration) : ((BotBus) -> Boolean) {
```

Comment:
I have added this example function here but I don't know if it's the right place